server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QualityGateConditionDto.java

```diff
@@ -206,4 +207,28 @@ public static boolean isOperatorAllowed(String operator, ValueType metricType) {
       return Collections.emptySet();
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
I don't like specifying equals() and hashCode() in DTOs. In your case, I think this logic should only be implemented in the RegisterQualityGates class.

Reply:
Right it was a remaining code that is useless